synapseclient/client.py

```diff
@@ -3124,14 +3125,17 @@ def _async_table_update(self, table, changes=[], create_snapshot=False,
 
         :return:  Snapshot Response
         """
-
+        snapshot_options = {'snapshotComment': comment,
+                            'snapshotLabel': label,
+                            'snapshotActivityId': activity}
+        new_snapshot = {key:value for key, value in snapshot_options.items() if value}
```

Comment:
`if value is not None` here since empty strings should probably be passed for these.